var/spack/repos/builtin/packages/py-mdanalysis/package.py

```diff
@@ -53,14 +56,21 @@ class PyMdanalysis(PythonPackage):
     depends_on("py-threadpoolctl", type=("build", "run"))
     depends_on("py-packaging", type=("build", "run"))
     depends_on("py-fasteners", type=("build", "run"))
-    depends_on("py-gsd@1.9.3:", type=("build", "run"))
+    depends_on("py-gsd@1.9.3:", when="@:2.5.0", type=("build", "run"))
 
     # extra_format (extras_require)
     depends_on("py-netcdf4@1.0:", when="+extra_formats", type=("build", "run"))
     depends_on("py-h5py@2.10:", when="+extra_formats", type=("build", "run"))
+    depends_on("py-pytng@0.2.3:", when="+extra_formats", type=("build", "run"))
     depends_on("py-chemfiles@0.10:", when="+extra_formats", type=("build", "run"))
     depends_on("py-pyedr@0.7.0:", when="+extra_formats", type=("build", "run"))
-    depends_on("py-pytng@0.2.3:", when="+extra_formats", type=("build", "run"))
+    #    depends_on("py-gsd@3.0.1:", when="+extra_formats @2.6.0:", type=("build", "run"))
```

Comment:
Probably deserves a comment explaining why it's commented out

Reply:
```suggestion
    # py-gsd is now an optional dependency and requires >3.0.0
    # gsd>=2.9.0 requires setuptools>=64 and can't be concretised with py-numpy
    #    depends_on("py-gsd@3.0.1:", when="+extra_formats @2.6.0:", type=("build", "run"))
```